---
layout: default
---

<div>
  This website accompanies the course <b>Data Science for High-Throughput Sequencing</b> (EE 372 at Stanford).  <br>
  For questions/comments/mistakes/typos in the course notes please leave a comment in the notes, or submit a pull request 
  directly to our <a href="https://github.com/data-science-sequencing/data-science-sequencing.github.io">git repo</a>.
  


</div>

<div class ="course-description">

   <div class="module-header">Course Description</div>

   Extraordinary advances in sequencing technology  in the past decade have revolutionized biology and medicine. Many high-throughput sequencing based assays have been designed to make various biological measurements of interest. This course explores the various computational and data science problems that arises from processing, managing and performing predictive analytics on high throughput sequencing data. Specific problems we will study include genome assembly, haplotype phasing, RNA-Seq assembly, RNA-Seq quantification, single cell RNA-seq analysis, multi-omics analysis, and genome compression. We attack these problems through a combination of tools from  information theory, combinatorial algorithms, machine learning and signal processing. Through this course, the student will also get familiar with various software tools developed for the analysis of real sequencing data.

</div>

<div class ="Lecture-times">

   <div class="module-header">Lectures times</div> 
    Monday, Wednesday 3:00 PM - 4:20 PM at 200-303
</div>

<div class ="course-staff">

   <div class="module-header">Course Staff</div> 

   Instructor: <a href="https://web.stanford.edu/~dntse/"> David Tse </a> (dntse _at_ stanford.edu) <br>
   Teaching assistants: <a href="https://web.stanford.edu/~gkamath"> Govinda Kamath </a> (gkamath _at_ stanford.edu) , <a href="https://web.stanford.edu/~jessez"> Jesse Zhang </a> (jessez _at_ stanford.edu) <br>
   Office hours to be decided.

</div>


<div class ="target-audience">

   <div class="module-header">Target Audience </div>
 
    <div class="materials-bullets">
    <ol type="1">
    <li>Students specializing in computational biology, who want to strengthen their knowledge of 
    basic information theory / signal processing / machine learning </li>
    <li>Students specializing in information theory / algorithms/ signal processing / 
    machine learning who want to learn of applications in biology</li>
    </ol> 
    </div>


  <div class="module-header">Pre-requisites </div>
    <div class="materials-bullets">
    <ul >
    <li>Undergraduate level probability </li>
    <li>Some programming experience. We will be using Python.</li>
    <li>Some undergraduate background in algorithms would be beneficial</li>
    <li>NO PRIOR background in biology will be assumed</li>
    </ul> 
  </div>
</div>


<div class ="grading">
  <div class="module-header">Course Grading (Tentative)</div>
  <div class="materials-bullets">
  <ul >
    <li>Attendance 10% </li>
    <li>Scribing 10% </li>
    <li>Assignments 40% </li>
    <li>Project 40%</li>
    </ul>
  </div>
    <br>
    There will be 4 assignments. The assignments will involve a theory component and a programming component.
    The programming component is aimed at exposing students to the messiness involved in real data, and various
    popular tools (so that they can gauge the pros and cons of these tools). The programming assignments will include 
    <div class="materials-bullets">
    <ul >
    <li>experiments demonstrating biases of different types in various types of data,</li>
    <li>implementing simple algorithms for assembly, alignment, and quantification,</li>
    <li>using popular software packages to perform simple experiments.</li>
    </ul>
    </div> 
    <br>
    All programming assignments will require require only laptop-level computing. The main language used to 
    code will be python. Familiarity with <a href="http://jupyter.org/"> Jupyter notebooks</a>, or willingness to
    learn would be beneficial. <br><br>
     UNIX/LINUX/osX may be needed for some of the
    software packages. We recommend that windows users use the <a href="https://web.stanford.edu/group/farmshare/cgi-bin/wiki/index.php/Main_Page"> 
    Corn cluster </a>.  <br><br>
    Projects can be theoretic or practical in nature (ideally
    a mix of the two), and can be done in groups of 2 or 3. A list of possible projects will be put up shortly. 
    Students can also come up with project topics that they are interested in (but are encouraged to check with 
    the course staff about the suitability of the project to this class).

</div>


<!-- <div class ="assignments">

  <div class="module-header">Assignments</div> 
  Assignment 1

</div> -->

<div class="home">
  <div class="materials-wrap">
    
    <div class="module-header">Course Materials</div>

    <div class="materials-item">
        <a href="/lectures/lecture0/">Course outline </a>
    </div>
 
  </div>
</div>
