---
layout: default
---

<div>
  This website accompanies the course <b>Data Science for High-Throughput Sequencing</b> (EE 372 at Stanford).  <br>
  For questions/comments/typos in the course notes please leave a comment in the notes or submit a pull request 
  directly to our <a href="https://github.com/data-science-sequencing/data-science-sequencing.github.io">Git repo</a>.
  


</div>

<div class ="course-description">

   <div class="module-header">Course Description</div>

   Extraordinary advances in sequencing technology  in the past decade have revolutionized biology and medicine. Many high-throughput sequencing based assays have been designed to make various biological measurements of interest. This course explores the various computational and data science problems that arises from processing, managing and performing predictive analytics on high throughput sequencing data. Specific problems we will study include genome assembly, haplotype phasing, RNA-Seq assembly, RNA-Seq quantification, single cell RNA-seq analysis, multi-omics analysis, and genome compression. We attack these problems through a combination of tools from  information theory, combinatorial algorithms, machine learning and signal processing. Through this course, the student will also get familiar with various software tools developed for the analysis of real sequencing data. The target audience for the course include
    
<div class="materials-bullets">
    <ol type="1">
    <li>Students specializing in information theory / algorithms / signal processing / 
    machine learning who want to learn of applications in biology and get exposure to real data</li>
    <li>Students specializing in computational biology, who want to strengthen their knowledge of 
    basic information theory / signal processing / machine learning </li>
    </ol> 
    </div>

<div class="module-subheader">Pre-requisites </div>
    <div class="materials-bullets-notopmargin">
    <ul >
    <li>Undergraduate level probability </li>
    <li>Some programming experience. We will be using Python.</li>
    <li>Some undergraduate background in algorithms would be beneficial</li>
    <li>No prior background in biology will be assumed</li>
    </ul> 
</div>

<div class ="Lecture-times">

   <div class="module-subheader">Lectures times</div> 
    Monday, Wednesday 3:00 PM - 4:20 PM at McCullough 115 <br>
    Lab hour: Friday (exact time and location TBA)
</div>

<div class ="course-staff">

   <div class="module-subheader">Course Staff</div> 

   Instructor: <a href="https://web.stanford.edu/~dntse/"> David Tse </a> (dntse _at_ stanford.edu) <br>
   Teaching assistants: <a href="https://web.stanford.edu/~gkamath"> Govinda Kamath </a> (gkamath _at_ stanford.edu) , <a href="https://web.stanford.edu/~jessez"> Jesse Zhang </a> (jessez _at_ stanford.edu) <br>
   Office hours: 4:20pm-5:05pm MW for instructor, 2:00pm-3:00pm M for teaching assistants

</div>

</div>

<div class ="grading">
  <div class="module-header">Course Grading (Tentative)</div>
  <div class="materials-bullets-addbotmargin">
  <ul >
    <li>Attendance 10% </li>
    <li>Scribing 10% </li>
    <li>Assignments 40% </li>
    <li>Project 40%</li>
    </ul>
  </div>
    
    <div class="module-subheader">Attendance</div>
    Students are encouraged to participate in class either during lecture or by leaving comments on material posted at the course website.
    
    <div class="module-subheader">Scribing</div>
    Each student will be responsible for scribing a lecture. To ensure that the notes will be available for students currently in the course, <b>scribed notes are due within 72 hours after lecture</b> (no late submissions accepted). Please use this <a href="https://docs.google.com/spreadsheets/d/1LkE4W4aFc7vi4FPkl4BeRQl9rD0pV1_CaGuqdqcHB38/edit#gid=0"> Google Doc </a> to reserve lectures for scribing. Scribed notes should be in either markdown or LaTeX. Scribes can either email the notes to the teaching assistants or submit a pull request at the course's Git repo.
    
    <div class="module-subheader">Assignments</div>
    There will be 4 assignments. The assignments will involve a theory component and a programming component.
    The programming component is aimed at exposing students to the messiness involved in real data and various
    tools used in practice. The programming assignments will include 
    <div class="materials-bullets-addtopbotmargin">
    <ul >
    <li>experiments demonstrating biases of different types in various types of data,</li>
    <li>implementing simple algorithms for assembly, alignment, and quantification,</li>
    <li>using popular software packages to perform simple experiments.</li>
    </ul>
    </div> 
    All programming assignments will require only laptop-level computing. The main language used to 
    code will be Python. UNIX/LINUX/OS X may be needed for some of the
    software packages. We recommend that windows users use the <a href="https://web.stanford.edu/group/farmshare/cgi-bin/wiki/index.php/Main_Page"> 
    Corn cluster</a>. 
    
    <div class="module-subheader">Projects</div>
    Projects can be theoretic or practical in nature (ideally a mix of the two). Additional details and a list of possible projects will be put up shortly. Students can also come up with project topics that they are interested in (in consultation with the teaching staff).

</div>


<!-- <div class ="assignments">

  <div class="module-header">Assignments</div> 
  Assignment 1

</div> -->

<div class="home">
  <div class="materials-wrap">
    
    <div class="module-header">Course Materials</div>

    <div class="materials-item">
        <a href="/handouts/Overview.pdf">Course overview </a>
        <a href="/lectures/lecture0/">Course outline </a>
    </div>
 
  </div>
</div>
